src/Sulu/Bundle/AdminBundle/Resources/js/components/Number/tests/Number.test.js

```diff
@@ -1,40 +1,44 @@
 // @flow
 import React from 'react';
-import {render, shallow} from 'enzyme';
+import {render, fireEvent, screen} from '@testing-library/react';
 import Number from '../Number';
 
 test('Number should render', () => {
-    expect(render(<Number onChange={jest.fn()} value={undefined} />)).toMatchSnapshot();
+    const {container} = render(<Number onChange={jest.fn()} value={undefined} />);
+    expect(container).toMatchSnapshot();
 });
 
 test('Number should render when disabled', () => {
-    expect(render(<Number disabled={true} onChange={jest.fn()} value={8} />)).toMatchSnapshot();
+    const {container} = render(<Number disabled={true} onChange={jest.fn()} value={8} />);
+    expect(container).toMatchSnapshot();
 });
 
 test('Number should call onChange with parsed value', () => {
     const onChange = jest.fn();
-    const number = shallow(<Number onChange={onChange} value={undefined} />);
+    render(<Number onChange={onChange} value={2} />);
 
-    const event = {};
-    number.find('Input').simulate('change', '10.2', event);
+    const input = screen.queryByDisplayValue(2);
+    fireEvent.keyDown(input, {target: {value: '10.2'}});
+
+    expect(input).toHaveValue(10.2);
 });
 
 test('Number should call onChange with undefined when value isn`t a float', () => {
     const onChange = jest.fn();
-    const number = shallow(<Number onChange={onChange} value={undefined} />);
+    render(<Number onChange={onChange} value={2} />);
 
-    const event = {};
-    number.find('Input').simulate('change', 'xxx', event);
+    const input = screen.queryByDisplayValue(2);
+    fireEvent.change(input, {target: {value: 'text'}});
 
-    expect(onChange).toBeCalledWith(undefined, event);
+    expect(onChange).toHaveBeenCalledWith(undefined, expect.anything());
 });
 
 test('Number should call onChange with undefined when value is undefined', () => {
     const onChange = jest.fn();
-    const number = shallow(<Number onChange={onChange} value={undefined} />);
+    render(<Number onChange={onChange} value={0.5} />);
 
-    const event = {};
-    number.find('Input').simulate('change', undefined, event);
+    const input = screen.queryByDisplayValue(0.5);
+    fireEvent.change(input, {target: {value: null}});
```

Comment:
Somehow I wasn't able to fire a change event with `value: undefined`. It just doesn't call the callback. Although it definitely should do so because of the `handleChange` logic in the component itself. But the event fired in the test won't even reach the component, at least I couldn't detect it with a `console.log`. Maybe this just doesn't count as *change* by RTL (which would be really weird). I now decided to pass `value: null` to the event, this works as expected. Is this problematic? I would really appreciate any help :)

Reply:
i think we should use the `clear` method of the `@testing-library/user-event` instead of firing the vent by ourselves 🙂

Reply:
Thanks!